### ISSUE TITLE
[ytdl_hook] Unable to stream Youtube videos

### ISSUE THREAD

**Issue by stevenxed** [mpv-player/mpv#5382]
### mpv version and platform
Windows 10 Home
mpv 0.28.0
### Reproduction steps
`mpv  AYoutubeURL`
### Expected behavior
Streams the video.
### Actual behavior
Cannot stream youtube video.
`[ytdl_hook] youtube-dl failed, trying to play URL directly ...`
### Log file
https://pastebin.com/7VGFeEKe
### notes
Youtube-dl itself can download the video.
Mpv is able to stream other sites besides Youtube.
Internet connection (wired) and speeds are not a problem.

**Comment by wiiaboo** [comment]
Works fine here. There were a couple fixes for youtube-dl on git master already. Could you try with that or at least give a log with `--msg-level=ytdl_hook=debug`?

**Comment by stevenxed** [comment]
Erh, could you tell me how to try those fixes? I'm new to all this.
Here's the log
`C:\Program Files\mpv> mpv --msg-level=ytdl_hook=debug https://www.youtube.com/watch?v=0_wtNwsAF3k
[ytdl_hook] Loading lua script @ytdl_hook.lua...
[ytdl_hook] loading mp.defaults
[ytdl_hook] loading @ytdl_hook.lua
[ytdl_hook] reading options for ytdl_hook
[ytdl_hook] lua-settings/ytdl_hook.conf not found.
Playing: https://www.youtube.com/watch?v=0_wtNwsAF3k
[ytdl_hook] found youtube-dl at: C:/Program Files/mpv/youtube-dl
[ytdl_hook] Running: C:/Program Files/mpv/youtube-dl --no-warnings -J --flat-playlist --sub-format ass/srt/best --no-playlist --format best --no-check-certificate --all-subs -- https://www.youtube.com/watch?v=0_wtNwsAF3k
[ytdl_hook] youtube-dl failed, trying to play URL directly ...
`

**Comment by wiiaboo** [comment]
That doesn't seem like a mpv issue, could you run that command manually?
`
"C:/Program Files/mpv/youtube-dl" --no-warnings -J --flat-playlist --sub-format ass/srt/best --no-playlist --format best --no-check-certificate --all-subs -- https://www.youtube.com/watch?v=0_wtNwsAF3k
`

**Comment by stevenxed** [comment]
I'm not sure if this is what u wanted:
`mpv "C:/Program Files/mpv/youtube-dl" --no-warnings -J --flat-playlist --sub-format ass/srt/bes
t --no-playlist --format best --no-check-certificate --all-subs -- https://www.youtube.com/watch?v=0_wtNwsAF3k
Error parsing option warnings (option not found)
Setting commandline option --no-warnings= failed.

Exiting... (Fatal error)`

**Comment by wiiaboo** [comment]
Remove "mpv" from the start.

**Comment by stevenxed** [comment]
I use Powershell normally, I've included the cmd logs too.
https://pastebin.com/w6kbbmyp

**Comment by wiiaboo** [comment]
Seems to work. Try with a newer build, maybe shinchiro's.

**Comment by stevenxed** [comment]
Okay, I've deleted the old mpv files and tried shinchiro's build (mpv-x86_64-20180110-git-0a406f9.7z).

C:\Program Files\mpv> mpv https://www.youtube.com/watch?v=0_wtNwsAF3k
Playing: https://www.youtube.com/watch?v=0_wtNwsAF3k
[ytdl_hook] youtube-dl failed
Failed to recognize file format.


Exiting... (Errors when loading file)

**Comment by wiiaboo** [comment]
Add `--log-file=whatever.txt` to mpv's command and then pastebin it.

**Comment by stevenxed** [comment]
https://pastebin.com/7CGYu5Vy

**Comment by wiiaboo** [comment]
Honestly have no idea. Try moving that whole mpv directory elsewhere out of "Program Files".

**Comment by wiiaboo** [comment]
And what's your youtube-dl version (run `youtube-dl --version`).

**Comment by stevenxed** [comment]
`youtube-dl --version`
`2018.01.07`
I've moved it to %appdata% (C:\Users\stevenxed\AppData\Roaming\mpv).
Same error in powershell.

**Comment by wiiaboo** [comment]
Save [this](https://i.fsbn.eu/XC1g.txt) as ytdl_hook.lua inside your mpv's config directory, inside a directory called `scripts` and run with `mpv --no-ytdl https://www.youtube.com/watch?v=0_wtNwsAF3k --msg-level=ytdl_hook=debug --log-file=whatever.txt` and pastebin the log again.

**Comment by stevenxed** [comment]
https://pastebin.com/RH2ZUMqh

**Comment by wiiaboo** [comment]
Well, there's some issue with whatever youtube-dl executable you're using. Can you try installing python 3.6, then install youtube-dl through it (`py -3 -m pip install youtube-dl`).
Delete youtube-dl.exe from mpv's config dir and then make sure python's youtube-dl is in PATH by using `where youtube-dl.exe`. If this failed, add the directory `Scripts` inside python's install directory to the system PATH.

If this still doesn't fix it, I have no idea.

**Comment by stevenxed** [comment]
How do I use python? Where do I paste that in.

**Comment by wiiaboo** [comment]
Powershell or cmd should be fine.

**Comment by stevenxed** [comment]
youtube-dl is in python's Scripts folder. 
I've added C:\Users\steve\AppData\Local\Programs\Python\Python36-32\Scripts into system PATH.
But running it again doesn't work.

`mpv https://www.youtube.com/watch?v=0_wtNwsAF3k`
```
Playing: https://www.youtube.com/watch?v=0_wtNwsAF3k
[ytdl_hook] youtube-dl failed
[ytdl_hook2] youtube-dl failed: not found or not enough permissions
[ytdl_hook3] youtube-dl failed: not found or not enough permissions
Failed to recognize file format.

Exiting... (Errors when loading file)
```

**Comment by Hrxn** [comment]
FYI, if you run the Python installer, all required default directories should already be added to PATH.

**Comment by wiiaboo** [comment]
Something is very wrong in your system or configuration.

If no one else can reproduce this, I'd say this is beyond this issue tracker.

**Comment by avih** [comment]
Can you please download the 0.28 from [here](https://mpv.srsfckn.biz/), extract it to some normal directory without spaces (e.g. `c:\testmpv\`), and then drag that youtube link onto the mpv.exe inside that dir? I'm thinking maybe there's more than one mpv or ytdl at the path which confuses things. And please use normal shell too (i.e. not power shell).

**Comment by Hrxn** [comment]
TBH, it should look like this:
`C:\Users\Hrxn\AppData\Local\Programs\Python\Python36`

**Comment by wiiaboo** [comment]
Only if it's not the 32-bit version.

**Comment by Hrxn** [comment]
Well, f.. me sideways, I think you're right.

**Comment by shebin007** [comment]
this problem can be solved by upgrading your youtube-dl(if you already have )
to upgrade use this
sudo pip install --upgrade youtube-dl
run it
enjoy

**Comment by VolatileCable** [comment]
Had the same issues and it was very strange indeed, I could only get it to work by installing youtube-dl from native windows python/pip. The official binary in the mpv folder (2020.06.16.1) wouldnt work and strangely enough would be skipped even though it's earlier in the search path. youtube-dl from cygwin pip also wouldnt work.

**Comment by wm4** [comment]
If you youtube-dl executable is just a python file with a shebang, it won't work, because we use CreateProcess, and Windows doesn't understand that.

**Comment by VolatileCable** [comment]
No, it's the official release from their website, it's some compiled python binary for windows. The pip install on cygwin is actually a python file and just imports the modules or whatever, while the windows pip install is again a native binary.

I don't really use that youtube-dl function otherwise, as it's kinda cumbersome, especially with the blanket ytdl-format often getting it wrong, so I don't care to investigate this further, but just thought I would share my experience here to maybe help others.

The error message is really undescript and I would imagine wrong, too:
```
[   0.786][e][ytdl_hook]  
[   0.786][e][ytdl_hook] youtube-dl failed: not found or not enough permissions 
```
